protocol/x/clob/memclob/memclob.go

```diff
@@ -375,7 +375,21 @@ func (m *MemClobPriceTimePriority) mustUpdateMemclobStateWithMatches(
 	}
 
 	// Add the new matches to the operations queue.
-	m.operationsToPropose.MustAddMatchToOperationsQueue(takerOrder, makerFillWithOrders)
+	internalOperation := m.operationsToPropose.MustAddMatchToOperationsQueue(takerOrder, makerFillWithOrders)
+
+	// If orderbook updates are on, send an orderbook update with the fill to grpc streams.
+	if m.generateOrderbookUpdates {
+		// Collect all maker orders.
+		makerOrders := lib.MapSlice(
+			makerFillWithOrders,
+			func(mfwo types.MakerFillWithOrder) types.Order {
+				return mfwo.Order
+			},
+		)
+		clobMatch := internalOperation.GetMatch()
+		orderbookMatchFill := m.GenerateOrderBookMatchFill(ctx, *clobMatch, takerOrder, makerOrders)
+		m.clobKeeper.SendOrderbookMatchFillUpdates(ctx, []types.OrderBookMatchFill{orderbookMatchFill})
+	}
```

Comment:
mega nit: these should be atomic, right? let's add a function to wrap these to be more future proof